pkg/cli/admin/ocpcertificates/monitorregeneration/secret.go

```diff
@@ -0,0 +1,144 @@
+			fmt.Fprintf(o.IOStreams.Out, "secrets/%v[%v] -- docker pull secret created\n", secret.Name, secret.Namespace)
+		}
+	}
+
+	o.interestingSecrets.upsert(secret.Namespace, secret.Name, secret)
+}
+
+func (o *MonitorCertificatesRuntime) updateSecret(obj, oldObj interface{}) {
+	secret, ok := obj.(*corev1.Secret)
+	if !ok {
+		fmt.Fprint(o.IOStreams.ErrOut, "unexpected update obj %T", obj)
+	}
+	defer o.interestingSecrets.upsert(secret.Namespace, secret.Name, secret)
+
+	oldSecret, ok := oldObj.(*corev1.Secret)
+	if !ok {
+		fmt.Fprint(o.IOStreams.ErrOut, "unexpected update oldObj %T", oldObj)
+	}
+
+	// we skip revisions because their information is not unique
+	if isForRevision(secret.OwnerReferences) {
+		return
+	}
+
+	o.handleTLSSecret(secret, oldSecret)
+}
+
+func (o *MonitorCertificatesRuntime) handleTLSSecret(secret, oldSecret *corev1.Secret) {
+	oldTLS, oldHasTLS := oldSecret.Data["tls.crt"]
+	newTLS, newHasTLS := secret.Data["tls.crt"]
+	if oldHasTLS && newHasTLS {
+		if !reflect.DeepEqual(oldTLS, newTLS) {
+			fmt.Fprintf(o.IOStreams.Out, "secrets/%v[%v] -- updated certificate, now expires at %v\n", secret.Name, secret.Namespace, secret.Annotations[certrotation.CertificateNotAfterAnnotation])
+		}
+	}
+
+	_, oldHasNotAfter := oldSecret.Annotations[certrotation.CertificateNotAfterAnnotation]
+	_, newHasNotAfter := secret.Annotations[certrotation.CertificateNotAfterAnnotation]
+	if oldHasNotAfter && newHasNotAfter {
+		oldRegenerating := oldSecret.Annotations[certrotation.CertificateNotAfterAnnotation] == "force-regeneration"
+		newRegenerating := secret.Annotations[certrotation.CertificateNotAfterAnnotation] == "force-regeneration"
+		switch {
+		case oldRegenerating && !newRegenerating:
+			fmt.Fprintf(o.IOStreams.Out, "secrets/%v[%v] -- finished regeneration\n", secret.Name, secret.Namespace)
+		case !oldRegenerating && newRegenerating:
+			fmt.Fprintf(o.IOStreams.Out, "secrets/%v[%v] -- started regeneration\n", secret.Name, secret.Namespace)
+		}
+	}
+}
+
+func (o *MonitorCertificatesRuntime) handleSAToken(secret, oldSecret *corev1.Secret) {
+	if isSAToken(secret) {
+		return
+	}
+
+	oldToken, oldHasToken := oldSecret.Data["token"]
+	newToken, newHasToken := secret.Data["token"]
+	switch {
+	case oldHasToken && !newHasToken:
+		fmt.Fprintf(o.IOStreams.Out, "secrets/%v[%v] -- serviceaccount token started regeneration\n", secret.Name, secret.Namespace)
+	case oldHasToken && newHasToken:
+		if !reflect.DeepEqual(newToken, oldToken) {
+			fmt.Fprintf(o.IOStreams.Out, "secrets/%v[%v] -- serviceaccount token updated\n", secret.Name, secret.Namespace)
+		}
+	case !oldHasToken && !newHasToken:
+	case !oldHasToken && newHasToken && len(newToken) > 0:
+		fmt.Fprintf(o.IOStreams.Out, "secrets/%v[%v] -- serviceaccount token finished regeneration\n", secret.Name, secret.Namespace)
+	}
+}
+
+func isSAToken(secret *corev1.Secret) bool {
+	if _, isSAToken := secret.Annotations["kubernetes.io/service-account.name"]; !isSAToken {
```

Comment:
should we just check the `Type`? https://github.com/kubernetes/kubernetes/blob/627f43d8bb6ce368fb7190aa8ffd303258766159/pkg/apis/core/types.go#L5456